### ISSUE TITLE
type-parameter-list-spacing not checked for generic extension properties

### ISSUE THREAD

**Issue by ZaMan0806** [pinterest/ktlint#3383]
## Expected Behavior
`standard:type-parameter-list-spacing` should require a single space before and after the type parameter list (`<T>`) of a generic **extension property**, the same way it already does for a generic extension **function**:

```kotlin
val <T> List<T>.head: T
    get() = this[0]
```

## Observed Behavior
For a generic extension property, missing spaces around the type parameter list are not detected at all:

```
$ ktlint --log-level=debug Head.kt
```

reports **no** `standard:type-parameter-list-spacing` violations for:

```kotlin
val<T>List<T>.head: T
    get() = this[0]
```

even though there is no space before `<T>` and no space after `<T>`. Running `ktlint -F` leaves the spacing untouched.

For comparison, the equivalent function extension is handled correctly:

```kotlin
fun <T>List<T>.head() = this[0]
```
```
$ ktlint Head2.kt
Head2.kt:1:8: Expected a single space (standard:type-parameter-list-spacing)
```

## Steps to Reproduce

```kotlin
val<T>List<T>.head: T
    get() = this[0]
```

Run `ktlint Head.kt` -> no violations reported (expected: 2 violations, one before and one after `<T>`).

I traced this to `TypeParameterListSpacingRule.kt`, function `beforeVisitChildNodes`. It dispatches on `node.parent?.elementType` with a `when` that only covers `CLASS`, `TYPEALIAS`, and `FUN` — there is no branch for `PROPERTY`, so the case falls through and the outer spacing (before/after the type parameter list) is never checked for generic extension properties. The spacing *inside* the angle brackets (`visitInsideTypeParameterList`) still runs regardless, since it isn't part of this `when`, so `<T,R>` vs `<T, R>` is still caught — only the outer spacing around `<...>` is missed for properties.

Uses default `.editorconfig` settings.

## Your Environment
* Version of ktlint used: `2.0.0-ALPHA-4` (also reproduced on latest `master`, commit `2b6028e4`)
* Relevant parts of the `.editorconfig` settings: default (`ktlint_official` code style)
* Name and version (or code for custom task) of integration used: ktlint CLI (locally built shadow jar)
* Version of Gradle used (if applicable): N/A (CLI usage)
* Operating System and version: macOS
